src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java

```diff
@@ -257,6 +263,26 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
         });
     }
 
+    @Override
+    public void init(FMLInitializationEvent e) {
+        super.init(e);
+
+        // Register IItemColor to handle passthrough
+        Minecraft.getMinecraft().getItemColors().registerItemColorHandler(new IItemColor() {
+            @Override
+            public int getColorFromItemstack(ItemStack stack, int tintIndex) {
+                CraftingPattern pattern = ItemPattern.getPatternFromCache(Minecraft.getMinecraft().world, stack);
+
+                if (GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1 &&
+                        Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex) != -1) {
```

Comment:
`Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex)` can probably go in another variable, we're reusing it twice.